docs/installation/install-binary.md

```diff
@@ -110,228 +106,703 @@ scp /usr/local/sbin/* karmada-02:/usr/local/sbin/
+│   │   ├── healthcheck-client.conf			# etcdctl => etcd
+│   │   ├── peer.conf										# etcd peer => etcd peer
+│   │   └── server.conf									# etcd server
+│   ├── front-proxy-client.conf					# kube-apiserver => karmada-aggregated-apiserver
+│   ├── karmada.conf										# most karmada component use this
+│   ├── kube-apiserver.conf							# tls server certificate of kube-apiserver
+│   └── kube-controller-manager.conf		# tls client & server cert of kube-controller-manager
+├── generate_ca.sh											# genererate front-proxy-ca, server-ca
+├── generate_etcd.sh    								# genererate CA & leaf certificates of etcd.
+├── generate_leaf.sh    								# generate leaf certificates of front-proxy-ca, server-ca
+└── util.sh
 ```
 
-#### Create root certificate
 
-valid for 10 years
+
+(1) generate_ca.sh
 
 ```bash
-openssl genrsa -out ca.key 2048
-openssl req -x509 -new -nodes -key ca.key -subj "/C=CN/ST=Guangdong/L=Guangzhou/O=karmada/OU=System/CN=karmada" -days 3650 -out ca.crt
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+function gen_server_ca() {
+	openssl genrsa -out server-ca.key 2048
+	openssl req -x509 -new -nodes -key server-ca.key -subj "/C=CN/ST=Guangdong/L=Guangzhou/O=karmada/OU=System/CN=karmada" -days 3650 -out server-ca.crt
+}
+
+function gen_front_proxy_ca() {
+	openssl genrsa -out front-proxy-ca.key 2048
+	openssl req -x509 -new -nodes -key front-proxy-ca.key -subj "/C=CN/ST=Guangdong/L=Guangzhou/O=karmada/OU=System/CN=front-proxy-ca" -days 3650 -out front-proxy-ca.crt
+}
+
+function main() {
+	mkdir ca_cert
+    cd ca_cert
+
+    gen_server_ca
+    gen_front_proxy_ca
+}
+
+main "$@"
 ```
 
-#### Create etcd certificate
 
-create `etcd server ` certificate
+
+(2) generate_etcd.sh
 
 ```bash
-openssl genrsa -out etcd-server.key 2048
-openssl req -new -nodes -sha256 -subj "/C=CN/ST=Guangdong/L=Guangzhou/O=karmada/OU=System/CN=karmada-etcd"  -key etcd-server.key -out etcd-server.csr
-openssl x509 -req -days 3650  \
-  -extfile <(printf "keyUsage=critical,Digital Signature, Key Encipherment\nextendedKeyUsage=serverAuth,clientAuth\nauthorityKeyIdentifier=keyid,issuer\nsubjectAltName=IP:172.31.209.245,IP:172.31.209.246,IP:172.31.209.247,IP:127.0.0.1,DNS:localhost") \
-  -sha256 -CA ca.crt -CAkey ca.key -set_serial 01 -in etcd-server.csr -out etcd-server.crt
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+source "./util.sh"
+
+# Absolute path to this script, e.g. /home/user/bin/foo.sh
+script="$(readlink -f "${BASH_SOURCE[0]}")"
+readonly script
+
+# Absolute path this script is in, thus /home/user/bin
+script_dir="$(dirname "$script")"
+readonly script_dir
+
+readonly csr_config_dir="${script_dir}/csr_config/etcd"
+readonly leaf_certs=(
+	"server"
+	"peer"
+	"healthcheck-client"
+	"apiserver-etcd-client"
+)
+
+function gen_etcd_ca() {
+	openssl genrsa -out ca.key 2048
+	openssl req -x509 -new -nodes -key ca.key -subj "/C=CN/ST=Guangdong/L=Guangzhou/O=karmada/OU=System/CN=etcd-ca" -days 3650 -out ca.crt
```

Comment:
refer to what you do before `openssl req -x509 -new -nodes -key server-ca.key -subj`, I prefer `openssl req -x509 -new -nodes -key etct-ca.key -subj xxxx`

Reply:
Well, I basically follow this document's recommended path: [Certificate paths](https://kubernetes.io/docs/setup/best-practices/certificates/#certificate-paths)

`server-ca` is the name used by the original document, I didn't renamed it.

Do you want me to strictly follow that document's recommended path?

And I move `apiserver-etcd-client.crt` from `/etc/karmada/pki/apiserver-etcd-client.crt` to `/etc/karmada/pki/etcd/apiserver-etcd-client.crt`. Because this way the certificates' path seem more related.

Reply:
Yeah, follow this path `/etc/karmada/pki/etcd/apiserver-etcd-client.crt` looks better.